x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ssl/SSLService.java

```diff
@@ -592,8 +593,15 @@ static Map<String, Settings> getSSLSettingsMap(Settings settings) {
         sslSettingsMap.put(WatcherField.EMAIL_NOTIFICATION_SSL_PREFIX, settings.getByPrefix(WatcherField.EMAIL_NOTIFICATION_SSL_PREFIX));
         sslSettingsMap.put(XPackSettings.TRANSPORT_SSL_PREFIX, settings.getByPrefix(XPackSettings.TRANSPORT_SSL_PREFIX));
         sslSettingsMap.putAll(getTransportProfileSSLSettings(settings));
-        // We build the ssl settings even when remote cluster port is not enabled because we may need it for the client side
-        sslSettingsMap.put(XPackSettings.REMOTE_CLUSTER_SSL_PREFIX, getRemoteClusterSslSettings(settings));
+        // Only build remote cluster server SSL if the port is enabled
+        if (REMOTE_CLUSTER_PORT_ENABLED.get(settings)) {
```

Comment:
tangential (and non blocking) to this PR: should we update this setting (and related) to include "server" and "client" to better match. i.e. `remote_cluster_server.enabled` and `remote_cluster_server.port`  ? (commenting on this line since the relevant file is not in the PR and probably outside the scope of this PR ... also even more tangentially, i think those setting would read better if prefixed with `xpack.security` such that you have a set of client settings and a set of server settings )

Reply:
Thanks for raising it. I was thinking about this as well. I think we should at least rename `remote_cluster.enabled` to `remove_cluster_server.enabled` because it does *not* control whether client is enabled. In fact, client is always enabled (but can be blocked because of missing `remote_cluster_client` role). As is, this setting is quite confusing because it seems to disable client as well which is not true. 

The port setting is less of an issue because it is relatively clear that this means for server. I do agree rename them both to include `server` explicitly look more consistent. But there are other settings in the same category as the `port`, e.g. `publish_port`, `host`, `publish_host`, etc. Does it feel a bit too much to rename them all? I think we can begin with renaming `remote_cluster.enabled` for now.

As for prefixing `xpack.security`, I agree that this would make settings look a lot more consistent as you suggested. But one wrinkle is that these settings are defined in the `server` package. None of existing settings in the server package has a `xpack` prefix. Is it OK to introduce it now? I can see three options:

1. Simply prefix the settings and do not move their locations. This has the implication of introducing `xpack.security` prefix into `server`. It also could be suprising that these are not *disabled* if `xpack.security.enabled` is `false`.
2. Prefix the settings and also *move* them to `x-pack/plugin/core`. Server has to read these settings with literal strings instead of the `Setting` class. This could be an issue if the settings have changed default or have validation rules. 
3. Similarly to 2 but also add support infrastructure so that server can get the values via dedicate Plugin method instead of retrieving them directly from `Settings`. It is quite a bit overhead for a rename. 

I'll take an action to raise a separate PR for rename just `remote_cluster.enabled` first. We can iterate how it should be done and what should be included over there. Thanks!